src/it/resources/org/checkstyle/suppressionxpathfilter/superfinalize/InputXpathSuperFinalizeMethodReference.java

```diff
@@ -0,0 +1,16 @@
+package org.checkstyle.suppressionxpathfilter.superfinalize;
+
+import com.puppycrawl.tools.checkstyle.checks.coding.superfinalize.InputSuperFinalizeMethodReference.CheckedConsumer;
+
+public class InputXpathSuperFinalizeMethodReference extends ClassWithFinalizer {
+    @Override
+    protected void finalize() throws Throwable {
+        CheckedConsumer<Throwable> r = super::finalize;
+        r.get();
+    }
+}
+
+class ClassWithFinalizer {
+    protected void finalize() throws Throwable { // warn
```

Comment:
From warning location and generated xpath both of this inputs are the same by structure.

Can we do second input to have finalize in inner class.

Please make third input for anonymous class.

Reply:
can you explain anonymous class as in?

Reply:
on it